### PR TITLE
Refactor vacuum clean to not need a separate status index on Task

### PR DESCRIPTION
**Pull request by CaseyHillers**
Instead, it will query the same amount of tasks that build status algorithm uses (25 at the moment). This will marginally increases the Datastore reads since it only run once an hour.

# Issues

https://github.com/flutter/flutter/issues/48914

### CODE REVIEW

File: app_dart/lib/src/request_handlers/vacuum-clean.dart [hunk 1]
```diff
@@ -90,6 +90,18 @@ class VacuumClean extends ApiRequestHandler<Body> {
...
   }
 
+  /// Returns whether [task] should be vacuum cleaned.
+  ///
+  /// The criteria are:
+  ///   1. [task] is currently marked as running.
+  ///   2. [task] has been running for at least an hour.
+  bool shouldBeVacuumCleaned(Task task) {
+    return _inProgress(task) && isOverAnHourOld(task);
+  }
```
Comment (digiter) [inline app_dart/lib/src/request_handlers/vacuum-clean.dart]: Nit: I wouldn't explain the criteria as the code is pretty self explanatory.

**Comment by CaseyHillers** [comment]
I used the Git web ui to resolve the merge conflict in index.yaml. It messed this commit up. I am going to rebase to fix it.

**Comment by godofredoc** [comment]
Casey can we merge this PR?

**Comment by CaseyHillers** [comment]
Yeah, I can baby sit this tonight watching to see if there's any datastore issues. If not, tomorrow morning i'll remove the index. Otherwise i'll do a revert.

**Comment by CaseyHillers** [comment]
No issues shown for the past 12 hours with the new refactor, I am going to remove this index and monitor throughout the day. Hoping not to break datastore on a Friday :)

**Comment by CaseyHillers** [comment]
```
gRPC Error (9, no matching index found. recommended index is:%0A- kind: Task%0A  properties:%0A  - name: Status%0A  - name: CreateTimestamp%0A    direction: desc%0A)
```
is showing only for the hourly vacuum clean job. Going to review the code today to see why it's still using that index.

**Comment by CaseyHillers** [comment]
Issue was just a mis deploy. I must have not gotten the top of master. Issue seems to be gone with Keyong's latest deploy. Will continue to keep an eye on it.

**Pull request merged**